src/node_messaging.cc

```diff
@@ -795,6 +795,11 @@ void MessagePort::OnMessage(MessageProcessingMode mode) {
 
   size_t processing_limit;
   if (mode == MessageProcessingMode::kNormalOperation) {
+    // Maybe the async handle was triggered empty or more than needed
+    // (called twiced on a new context).
+    if (GetTransferMode() == TransferMode::kDisallowCloneAndTransfer || !data_)
```

Comment:
Isn’t that just equivalent to this?

```suggestion
    if (IsDetached())
```

Reply:
That's right, changing it